examples/question-answering/run_qa.py

```diff
@@ -490,7 +533,15 @@ def post_processing_function(examples, features, predictions):
             ]
         else:
             formatted_predictions = [{"id": k, "prediction_text": v} for k, v in predictions.items()]
-        references = [{"id": ex["id"], "answers": ex[answer_column_name]} for ex in datasets["validation"]]
+
+        # if we are using post processing for test stage we will use respective example
+        if stage == "test":
+            examples = test_examples
+        else:
+            examples = eval_examples
```

Comment:
I don't think those four lines are necessary: I believe we can use the `examples` variable passed. It will be what was given to the `Trainer` as `eval_examples` (if in `evaluate`) or `test_examples` (if in `predict`).